src/connectors/impls/pulsar/client.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use dashmap::DashMap;
+use pulsar::{Pulsar, TokioExecutor};
+use tokio::sync::Mutex;
+
+use crate::connectors::prelude::*;
+
+lazy_static! {
+    static ref SINGLETON_LOCK: Mutex<()> = Mutex::new(());
+    static ref PULSAR_CLIENT_SINGLETON: DashMap<String, Pulsar<TokioExecutor>> = DashMap::new();
+}
+
+// TODO: Add other possible configurations
+pub(crate) async fn get_client(url: &str) -> Result<Pulsar<TokioExecutor>> {
```

Comment:
Hey, I will remove.
I thought that one client for each topic would be too much because as they would have to do the tls handshake, authentication, and authorization.

Reply:
The way we handle that wtih kafka is allowing a single kafka connector to produce to multiple topics but since they're long lived I don't think the overhead of a extra tls handshake for a topic is too bad?